pkg/srvdiscovery/interface.go

```diff
@@ -17,18 +17,23 @@ type ServiceResource = model.ExecutorInfo
 
 // WatchResp defines the change set from a Watch API of Discovery interface
 type WatchResp struct {
-	addSet map[UUID]ServiceResource
-	delSet map[UUID]ServiceResource
-	err    error
+	AddSet map[UUID]ServiceResource
+	DelSet map[UUID]ServiceResource
+	Err    error
 }
 
 // Discovery defines interface of a simple service discovery
 type Discovery interface {
 	// Snapshot returns a full set of service resources and the revision of snapshot
-	// updateCache indicates whether to update the snapshot to Discovery cache.
-	Snapshot(ctx context.Context, updateCache bool) (map[UUID]ServiceResource, Revision, error)
+	Snapshot(ctx context.Context) (map[UUID]ServiceResource, error)
 
 	// Watch watches the change of service resources, the watched events will be
 	// returned through a channel.
 	Watch(ctx context.Context) <-chan WatchResp
+
+	// SnapshotClone returns the clone of cached snapshot in Discovery
+	SnapshotClone() (map[UUID]ServiceResource, Revision)
+
+	// CopySnapshot copies snapshot from given Discovery
+	CopySnapshot(snapshot map[UUID]ServiceResource, revision Revision)
```

Comment:
Probably we should comment on when `CopySnapshot` is useful?

Reply:
added comment